crates/bevy_xr_utils/src/tracking_utils.rs

```diff
@@ -0,0 +1,314 @@
+            //TODO: use yaw
+            let (y, x, z) = calc_floor.rotation.to_euler(EulerRot::YXZ);
+            let new_rot = Quat::from_rotation_y(y);
+            calc_floor.rotation = new_rot;
+            for (mut transform) in &mut local_floor {
+                *transform = calc_floor;
+            }
+        }
+        Err(_) => (),
+    }
+}
+//external
+fn update_local_floor(
+    mut local_floor: Query<&mut Transform, (With<LocalFloor>, Without<XRTrackedLocalFloor>)>,
+    mut tracked_floor: Query<&mut Transform, (With<XRTrackedLocalFloor>, Without<LocalFloor>)>,
+) {
+    let head_transform = local_floor.get_single_mut();
+    match head_transform {
+        Ok(head) => {
+            for (mut transform) in &mut tracked_floor {
+                *transform = head.clone();
+            }
+        }
+        Err(_) => (),
+    }
+}
+
+//left grip
+#[derive(Component)]
+struct LeftGrip;
+
+fn update_left_grip(
+    mut left_grip: Query<&mut Transform, (With<LeftGrip>, Without<XRTrackedLeftGrip>)>,
+    mut tracked_left_grip: Query<&mut Transform, (With<XRTrackedLeftGrip>, Without<LeftGrip>)>,
+) {
+    let head_transform = left_grip.get_single_mut();
+    match head_transform {
+        Ok(head) => {
+            for (mut transform) in &mut tracked_left_grip {
+                *transform = head.clone();
+            }
+        }
+        Err(_) => (),
+    }
+}
+
+//right grip
+#[derive(Component)]
+struct RightGrip;
+
+fn update_right_grip(
+    mut right_grip: Query<&mut Transform, (With<RightGrip>, Without<XRTrackedRightGrip>)>,
+    mut tracked_right_grip: Query<&mut Transform, (With<XRTrackedRightGrip>, Without<RightGrip>)>,
+) {
+    let head_transform = right_grip.get_single_mut();
+    match head_transform {
+        Ok(head) => {
+            for (mut transform) in &mut tracked_right_grip {
+                *transform = head.clone();
+            }
+        }
+        Err(_) => (),
+    }
+}
+
+//tracking rig
+#[derive(Resource)]
+struct ControllerActions {
+    set: openxr::ActionSet,
+    left: openxr::Action<Posef>,
+    right: openxr::Action<Posef>,
+}
+
+fn spawn_tracking_rig(
+    actions: Res<ControllerActions>,
+    mut cmds: Commands,
+    root: Query<Entity, With<XrTrackingRoot>>,
+    session: Res<OxrSession>,
+) {
+    //head
+    let head_space = session
+        .create_reference_space(openxr::ReferenceSpaceType::VIEW, Transform::IDENTITY)
+        .unwrap();
+    let head = cmds
+        .spawn((SpatialBundle { ..default() }, HeadXRSpace(head_space)))
+        .id();
+    let local_floor = cmds.spawn((SpatialBundle { ..default() }, LocalFloor)).id();
+
+    let left_space = session
+        .create_action_space(&actions.left, openxr::Path::NULL, XrPose::IDENTITY)
+        .unwrap();
+    let right_space = session
+        .create_action_space(&actions.right, openxr::Path::NULL, XrPose::IDENTITY)
+        .unwrap();
+    let left = cmds
+        .spawn((SpatialBundle { ..default() }, left_space, LeftGrip))
```

Comment:
any reason why you don't just use `SpatialBundle::default()`?

Reply:
for some reason that was throwing errors.